File: cli/internal/client/filewatchers.go
package client

import (
	"context"
	"net/url"
)

// ListFileWatchers returns the active profile's file watcher subscriptions.
func (c *Client) ListFileWatchers(ctx context.Context) ([]map[string]any, error) {
	var resp struct {
		Subscriptions []map[string]any `json:"subscriptions"`
	}
	if err := c.GetJSON(ctx, "/api/file-watchers", &resp); err != nil {
		return nil, err
	}
	return resp.Subscriptions, nil
}

// DeleteFileWatcher removes a single file watcher subscription.
func (c *Client) DeleteFileWatcher(ctx context.Context, id string) error {
	return c.Delete(ctx, "/api/file-watchers/"+url.PathEscape(id), nil)
}

// CreateFileWatcher registers a new file watcher subscription.
//
// `body` follows the schema accepted by POST /api/file-watchers — see the
// Python tool ``register_file_watcher`` for fields (path, name, triggers,
// target_kind, extensions, recursive, action, conversation_id).
func (c *Client) CreateFileWatcher(ctx context.Context, body map[string]any) (map[string]any, error) {
	var out map[string]any
	if err := c.PostJSON(ctx, "/api/file-watchers", body, &out); err != nil {
		return nil, err
	}
	return out, nil
}

// FileWatchersAdminStreamPath returns the path for the admin SSE snapshot.
func (c *Client) FileWatchersAdminStreamPath() string {
	return "/api/file-watchers/admin/stream"
}

File: app/cli/client/file_watchers.py
"""File-watcher endpoints — `/api/file-watchers*`.

Mirrors `cli/internal/client/filewatchers.go`.
"""

from __future__ import annotations

from typing import Any
from urllib.parse import quote

from app.cli.client._base import Client


async def list_file_watchers(client: Client) -> list[dict[str, Any]]:
    resp = await client.get_json("/api/file-watchers")
    if isinstance(resp, dict) and isinstance(resp.get("subscriptions"), list):
        return [s for s in resp["subscriptions"] if isinstance(s, dict)]
    return []


async def delete_file_watcher(client: Client, watcher_id: str) -> None:
    await client.delete(f"/api/file-watchers/{quote(watcher_id, safe='')}")


async def create_file_watcher(
    client: Client,
    body: dict[str, Any],
) -> dict[str, Any]:
    resp = await client.post_json("/api/file-watchers", body)
    return resp if isinstance(resp, dict) else {}


def file_watchers_admin_stream_path() -> str:
    return "/api/file-watchers/admin/stream"
